lsp/pylyzer.lua

```diff
@@ -22,4 +28,7 @@ return {
       checkOnType = false,
     },
   },
+  on_new_config = function(new_config, _)
```

Comment:
no such thing in vim.lsp.config. try `before_init` ? and please test that it actually works.

Reply:
Sure, though that means the CWD "sticks" and won't dynamically update. I don't use this config so I have no strong opinion :)

Reply:
I mean it wasn't really doing anything dynamic anyway. It was essentially just setting `ERG_PATH=~/.erg`. Unless I'm missing something...?

And `before_init` happens after starting the binary. So setting `cmd_env` wouldn't have any effect?

Reply:
> `before_init` happens after starting the binary

oh. Then `cmd` should be defined as a function (again, assuming this functionality is important for users of this config)

Reply:
> > `before_init` happens after starting the binary
> 
> 
> 
> oh. Then `cmd` should be defined as a function (again, assuming this functionality is important for users of this config)

I'm not sure about this though. Just going by the docs:

> Callback invoked before the LSP "initialize" phase, where params contains the parameters being sent to the server and config is the config that **was** passed to vim.lsp.start(). You can use this to modify parameters before they are sent.

Seems to indicate that `vim.lsp.start()` has already been called.

Reply:
With that said. I suppose `cmd` being a function could be useful when you want to start a server relative to the resolved root.

For example you might want to call `./node_modules/.bin/biome` relative to that root.

Though I suppose that could also be solved by having relative paths be relative to workspace root instead of cwd (though relative paths aren't supported at all at the moment I think?). 

Or do the VS Code way (or CoC, don't remember which). By using some magic marker like `<root_dir>/node_modules/.bin/biome`.